docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,113 @@
+# ARCHITECTURE
+
+An `ActiveModel::Serializer` is essentially a [Decorator](http://c2.com/cgi/wiki?DecoratorPattern)
+for a resource. It allows you to specify which attributes and
+associations should represent the serialization of the resource. It requires an adapter to
+transform its attributes into a JSON document; it cannot be serialized itself.
+
+The `ActiveModel::ArraySerializer` represent a collection of resources as serializers
```

Comment:
ref https://github.com/rails-api/active_model_serializers/pull/1251 to rename it to CollectionSerializer